brownie/project/sources.py

```diff
@@ -211,9 +211,13 @@ def get_pragma_spec(source: str, path: Optional[str] = None) -> NpmSpec:
     Returns: NpmSpec object
     """
 
-    pragma_match = next(re.finditer(r"pragma +solidity([^;]*);", source), None)
+    pragma_match = next(re.finditer(r"(?<!([\*/ \t]))pragma +solidity([^;]*);", source), None)
```

Comment:
A better fix would be to match on the start of line, i.e. `^pragma...`

Reply:
I just tested this revision locally and using this method, I get 3 failing tests:

FAILED tests/project/compiler/test_solidity.py::test_build_json_keys - brownie.exceptions.PragmaError: No version pragma in 'path.sol'
FAILED tests/project/compiler/test_solidity.py::test_size_limit - brownie.exceptions.PragmaError: No version pragma in 'foo.sol'
FAILED tests/project/compiler/test_solidity.py::test_find_best_solc_version - brownie.exceptions.PragmaError: No version pragma in '<stdin>'

Reply:
since that didn't work, I tried some other varieties and found that this logic used for vyper also works for sol:
(?:\n|^)pragma +solidity([^;]*);

I've updated my branch accordingly